app/models/set_member_subject.rb

```diff
@@ -48,4 +58,11 @@ def remove_from_queues
   def set_random
     self.random = rand
   end
+
+  def timestamp_newly_retired_workflows
+    retired_subject_workflow_counts.each do |record|
+      record.retired_at ||= Time.now
+      record.save if record.persisted? and record.changed?
```

Comment:
Loosk like this would be better as a method on `retired_subject_workflow_count`, e.g. `record.update_retired!`.